### ISSUE TITLE
Updates to Equipments

### ISSUE THREAD

**Issue by lordcygnus** [BeeGrinder/FantasyGrounds-SW5e#35]
As before, we need to update to the most recent version of all equipment present on the PHB.

**Comment by lordcygnus** [comment]
I've just finished up building the equipment file for the update, this one has all equipment present on the PHB, apart from the Droids and Trade Goods (those two groups have issues with the par5e, so, if you can make them available on your end, highly appreciated).
[equipment.txt](https://github.com/BeeGrinder/FantasyGrounds-SW5e/files/5648548/equipment.txt)

**Comment by BeeGrinder** [comment]
Ok, this is where things get weird with 5e, sw5e, and par5e.  Particularly with weapons.  I was never able to get these working correctly in par5e.
The problem is in the Item button list, getting the buttons at the top for armor and weapons to work correctly.  (I've never even tried the forge or template options).  But, it's actually not par5e's fault.  FG is basically a text parser that is trained to interpret the phrasings of WoTC's standardized text.  If you've ever read the document from WoTC, they are VERY particular about phrasing.  This is because they know systems like FG parse their phrases into game tools and actions.

There are two options to fix SW5e phrasing to work in the 5e ruleset:
1.  mimic WoTC phrasing.
2.  change the Lua code to interpret SW5e phrasing.

For me so far, choice one works more reliably.  Which is why you see my vibroweapons being listed as melee weapons.  This is also used in the combat tracker to distinguish ranged from melee attacks when clinking on the to-hit and damage buttons.  The combat tracker and character sheets don't know what a lightsaber or vibroweapon is.  So, I kept vibroweapons as melee and blasters as ranged.  I added the category for simple/martial "lightsaber weapons" because I found without the word "weapons" some things failed.  But, I also don't know how well these work in the combat tracker.  I think since it seems to default to melee, it works fine.

I think mundane non-weapons (like rope, glowstick, datapad, etc) work fine in par5e.  But, anything with enhanced properties or weapons were causing me so much grief I ended up hand-coding them to XML in the supplement.  In fact, I just did all items that way.

These are areas that made me wish I'd not based sw5e on the 5e ruleset.  Building up from CoreRPG (or even MoreCore) ruleset would have been easier to maintain in the end.

Sorry for the wall-of-text.
TLDR; items/equipment have been troublesome and I ended up not using par5e to make it work with the FG ruleset code.

**Comment by BeeGrinder** [comment]
After looking, I still think the way the xml is setup today is the best bet to have these items in a working state.

For updates, a list of delete, modify, add change-list would be most efficient.  Editing the XML wholesale is a LOT of work as each item has an id# and a list reference with that same id# so it all has to match.  With a change list, it can be managed and id#s can be recycled if deleted.

same update strategy for starting packs (parcels).

**Comment by lordcygnus** [comment]
I've used PAR5E to make the txt file, and tested it with a created TEST module on my end, and it worked all fine, didn't know that it will cause so much trouble behind the scenes, I was aware of the ranged/melee weapon situation, but it didn't matter much to me thanks to the character sheet changing the type of weapon referenced with a simple click on the Sword/Bow-Melee/Ranged button on the actions, aside from that, for references with the classes, also thought that would be a better idea to have the weapons separated by their original subtype (#st;), but just let me know if you need anything on my end to help out with adding the items on the list, in that txt file I've sent are all the equipment in the PHB, most of them with added descriptions with the explanation of their properties (in case of the armors and weapons).

**Comment by BeeGrinder** [comment]
This is encouraging.  The weapon categories are correct.  In version 3.x of FG having anything other than 5e verbiage really messed up those armor/weapon buttons in the item window.

**Comment by BeeGrinder** [comment]
Great work!  I have not found anything going wrong after altering the Lua code for the old categories.  They definitely changed something. So, now I have to remove all items from the spell supplement module.  Which is great since it will be easier to maintain.

Would you like to see this module?  I am doing all these changes in a separate branch since it's both modules and the extension all at once.  I can check the test branch in to gitlab for you to test before I merge this all into the main branch.

**Comment by lordcygnus** [comment]
Cool, happy to know it worked somehow lol just post it when ready, I have the equipment.txt for the EC and other SW5E books on another built module, better to just leave the PHB stuff on yours for easy maintenance related work.

**Comment by BeeGrinder** [comment]
I am getting a strange console error. "[ERROR] Object (DATABASENODE) deleted without cleaning up attached Lua object."

This only happens if you only load the PHB module and then open up the items list. then exit by returning to the main "load campaign" screen.  If I remove the equipment.txt loading in par5e this does not occur.  I'll need to spend some time figuring out what might be causing this.  Looking at the forums, it is a new error created to find problematic modules.

**Comment by lordcygnus** [comment]
Great, let me know if I can be of any help, going to continue on my thing (reviewing Races and Spells now)

**Comment by BeeGrinder** [comment]
Ok, just remember, the reason all the spells are in XML is because of all the effects coding.  I used to have the spells in par5e format, but most did not go into the combat tracker at all.  And those that did were often not correct.  I probably spent as much time fixing "spells" as I have on the entire rest of the project.

The only way I can figure to do "powers" in par5e is to reword them exactly like their WoTC wording, which is probably impossible due to their modifications and reflavoring for sw5e.

**Comment by lordcygnus** [comment]
send me the xml file for the spells, probably It's going to be way faster than me doing a .txt version of it and then you figuring out what to do with the effects,

**Comment by BeeGrinder** [comment]
it is this one https://github.com/BeeGrinder/FantasyGrounds-SW5e/blob/master/supplement-mod/common.xml
But, it has this format. Each spell has 2 entries with identical id-#
<root><lists><power<index> see here each is a reference to a spell.  name must match the name property of the true spell entry.

<root>spell>... the actual spell entries.  The <actions> section is where all the combat tracker stuff comes from.

These can be a bit overwhelming.  If its easier for you to create a list of adds/deletes/changes by name, then I can just look them up in the PDF  and do it.  I have done a bunch of them as you can see.  So, feel free to go that route.

**Comment by BeeGrinder** [comment]
I'm closing this one and opening an equipment audit project and an equipment reference project
